pkg/process/process.go

```diff
@@ -0,0 +1,90 @@
+package process
+
+import (
+	"os"
+	"strconv"
+	"strings"
+	"syscall"
+
+	"github.com/pkg/errors"
+)
+
+type Manager interface {
+	Pid() (int, error)
+	Name() string
+	PidFilePath() string
+	Exists() bool
+	Kill() error
+	FindProcess() (*os.Process, error)
+	WritePidFile(pid int) error
+}
+
+type Process struct {
+	name        string
+	pidFilePath string
+}
+
+func New(name, pidFilePath string) (*Process, error) {
+	return &Process{name: name, pidFilePath: pidFilePath}, nil
+}
+
+func (p *Process) Name() string {
+	return p.name
+}
+
+func (p *Process) PidFilePath() string {
+	return p.pidFilePath
+}
+
+func (p *Process) Pid() (int, error) {
+	data, err := os.ReadFile(p.PidFilePath())
+	if err != nil {
+		return -1, err
+	}
+	pidStr := strings.TrimSpace(string(data))
+	pid, err := strconv.Atoi(pidStr)
+	return pid, errors.Wrap(err, "invalid pid file")
```

Comment:
`errors.Wrap` is deprecated, https://github.com/pkg/errors is archived. This has been superseded by https://go.dev/blog/go1.13-errors